colorizer_data/types.py

```diff
@@ -97,15 +100,64 @@ class FrameDimensions(TypedDict):
     """Height of a frame in physical units (not pixels)."""
 
 
-class DatasetMetadata(TypedDict):
+class BaseMetadataJson(TypedDict):
+    """JSON dictionary format for `BaseMetadata`."""
+
+    name: str
+    description: str
+    author: str
+    dateCreated: str
+    lastModified: str
+    revision: str
+    dataVersion: str
+
+
+@dataclass
+class BaseMetadata:
+    """Shared metadata between datasets and collection files."""
+
+    name: str = None
+    description: str = None
+    author: str = None
+    date_created: str = None
+    """Formatted datetime string. See `DATETIME_FORMAT`. """
+    last_modified: str = None
```

Comment:
I'm assuming that tooling will handle date_created and date_modified without users having to touch them explicitly?  (While allowing them to provide the data if they feel like it)

Reply:
Yup! I have a working prototype of this on a different branch.